### PR TITLE
Pim 5666 issue 4179

### PR DESCRIPTION
**Pull request by a2xchip**
**Description (for Contributor and Core Developer)**

Fix for #4179 and PIM-5666

**Definition Of Done**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | Y
| Changelog updated                 | -
| Review and 2 GTM                  | -

### CODE REVIEW

**Comment by a2xchip** [comment]
@nidup @jjanvier @juliensnz Hello! I found the way to fix issue #4179 and waiting for your reviews. Is it GTM :pray: :pray: :pray:??

File: src/Pim/Component/Catalog/Comparator/Attribute/NumberComparator.php [hunk 1]
```diff
@@ -48,6 +48,10 @@ public function compare($data, $originals)
             return $data;
         }
 
+        if ((string) $data['data'] == '0' && null === $originals['data']) {
```
Comment (juliensnz) [inline src/Pim/Component/Catalog/Comparator/Attribute/NumberComparator.php]: We tend to use the Yoda style for if assertions. Is it possible to switch `$data['data']` and `'0'`

**Comment by juliensnz** [comment]
Really nice, thanks a lot for this contribution!

GTM for me after small fix.

**Comment by pierallard** [comment]
Thanks again @a2xchip for your contributions :gift: ! Same comment than @juliensnz , then it's GTM for me too!

File: src/Pim/Component/Catalog/Comparator/Attribute/NumberComparator.php (hunk 1 above)
Comment (a2xchip) [inline src/Pim/Component/Catalog/Comparator/Attribute/NumberComparator.php]: Oops. Do you mean ```'0' == $data['data']```? I thought you use such rules only for null comparisons. I'll update it soon! :-)

**Comment by a2xchip** [comment]
@pierallard @juliensnz Thank you! I'update it soon! And look for another open issue!!! :smile: :smile:

**Comment by willy-ahva** [comment]
Hey @a2xchip !

I'm very sorry but this bug has already been fixed in the v1.5 but not already merged.
You can find it here : https://github.com/akeneo/pim-community-dev/pull/4293
Due to a freeze of the 1.5 branch we are not able to merge it before 1 or 2 day(s).

This fix will be available in master when the next 1.5 will be tagged and when we'll merge 1.5 into master.

Sorry again, maybe when we work on a fixed we should update the corresponding issue in Github when it exists. We'll discuss about that.

Thank you for contributing, your work is amazing. By the way, I sent you an email to get your postl address to thank you, maybe you missed it. In this case you can directly contact me be email.

Willy

**Comment by a2xchip** [comment]
@willy-ahva Damn! The same PR as mine ((( 15 days ago...

**Comment by pierallard** [comment]
Hey @a2xchip !
Just merged @willy-ahva 's PR. I close this one, nice try btw ;)

**Pull request closed without merging**